packages/database/src/changeHandlers/TaskCompletionHandler.js

```diff
@@ -57,23 +57,37 @@ export class TaskCompletionHandler extends ChangeHandler {
       },
     });
 
-    return tasks.map(task => task.id);
+    const mappedTasksToSurveyResponses = {};
+
+    // map the task ids to the survey response ids
+    tasks.forEach(task => {
+      const { survey_id: surveyId, entity_id: entityId, created_at: createdAt, id } = task;
+      const matchingSurveyResponse = surveyResponses.find(
```

Comment:
If I am understand this correctly, then one survey response could "complete" multiple tasks which means that there is an assumption that users don't create multiple tasks for the same survey and entity for the same time period. Is that right?

Reply:
There could be multiple survey responses that 'complete' a task, and one survey response could 'complete' multiple tasks. There's no restriction on how many tasks can be related the same survey/entity/time period. However we only save 1 survey response ID to a task, so if there are multiple, then we would save the first one in the results.